pkg/daemons/config/types.go

```diff
@@ -290,3 +293,46 @@ func GetArgs(argsMap map[string]string, extraArgs []string) []string {
 	sort.Strings(args)
 	return args
 }
+
+// HashConfig creates a hash with important config that must be shared across control plane members of the cluster
+func HashConfig(server *Control) string {
+	var configData []string
+
+	// Add ClusterIPRanges and ServiceIPRanges
+	ipSlice := append(server.ClusterIPRanges, server.ServiceIPRanges...)
+	for _, ipRange := range ipSlice {
+		configData = append(configData, ipRange.String())
+	}
+
+	// Add ClusterDNSs
+	for _, ClusterDNS := range server.ClusterDNSs {
+		configData = append(configData, ClusterDNS.String())
+	}
+
+	// Add ClusterDomain and FlannelBackend
+	configData = append(configData, server.ClusterDomain, server.FlannelBackend)
+
+	// Check disables[] and skips[]
+	for key, value := range server.Disables {
+		if value {
+			configData = append(configData, key)
+		}
+	}
+
+	for key2, value2 := range server.Skips {
+		if value2 {
+			configData = append(configData, key2)
+		}
+	}
+
+	// Finally add the boolean fields
+	moreFields := []bool{server.DisableCCM, server.DisableNPC, server.DisableHelmController, server.DisableKubeProxy, server.DisableAPIServer, server.DisableControllerManager, server.DisableScheduler, server.DisableETCD, server.NoCoreDNS, server.DisableServiceLB}
+
+	for _, boolField := range moreFields {
+		configData = append(configData, strconv.FormatBool(boolField))
+	}
+
+	// hash everything
+	hash := sha256.Sum256([]byte(strings.Join(configData, "")))
+	return base64.StdEncoding.EncodeToString(hash[:])
+}
```

Comment:
Is there any way we can somehow indicate at the flag/config level that we want to/do not want to include th various fields for hashing? The task to keep this hashing function up to date seems like a bit of tech debt/tribal knowledge to me

Reply:
agree. I should find a better way

Reply:
Alternatively, if we just add comments to make sure to check this where the config flags/config structs are defined that would probably work to eliminate the tribal knowledge side.

Reply:
I found a good way to fix this by using anonymous fields